### PR TITLE
Fix bugs and documentation for `plot_backtest` and `plot_backtest_interactive`

### LINKED ISSUE #664: Some fixes for `plot_backtest`
### 🚀 Feature Request

Fix some issues in`plot_backtest`

### Motivation

Make eda method better

### Proposal

1. Add note about the `history_len=-1` in the description, it should plot all the history(fix it if it doesn't)
2. Fix the behavior of parameter `folds`
    - Don't draw the forecast/test after the last fold(e.g. we got `folds=[1]` and out plot ends after the end of the first fold)
3. Fix the behavior when the folds were created using masks(or not sequential folds are passed to the folds parameter e.g. `folds=[1,3,5]`):
    - Draw train part between the folds(now it draws the long test part instead)
    - Add note to the description that it is better to select the fold to see the backtest results in details
 4. Parameter `history_len` should work only for the first fold, we do not count the history between the folds

### Test cases

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [ ] I discussed this issue with ETNA Team

### PR DESCRIPTION
**Pull request by Mr-Geekman**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes
Look #664.

## Related Issue
#664.

## Closing issues
Closes #664.

### CODE REVIEW

**Comment by Mr-Geekman** [comment]
Script for `plot_backtest`:
```python
import matplotlib.pyplot as plt
import pandas as pd

from etna.analysis import plot_backtest
from etna.datasets import TSDataset
from etna.metrics import MAE
from etna.models import ProphetModel
from etna.pipeline import FoldMask
from etna.pipeline import Pipeline


def main():
    df = pd.read_csv("examples/data/example_dataset.csv")
    df = TSDataset.to_dataset(df)
    ts = TSDataset(df, freq="D")

    pipeline = Pipeline(model=ProphetModel(), horizon=14)
    metrics_df, forecast_df, fold_info_df = pipeline.backtest(ts=ts, metrics=[MAE()], n_folds=3)

    # no history
    plot_backtest(forecast_df=forecast_df, ts=ts, history_len=0)
    plt.savefig("plot_no_history.png")

    # partial history
    plot_backtest(forecast_df=forecast_df, ts=ts, history_len=50)
    plt.savefig("plot_partial_history.png")

    # all history
    plot_backtest(forecast_df=forecast_df, ts=ts, history_len="all")
    plt.savefig("plot_all_history.png")

    # fold mask example
    pipeline = Pipeline(model=ProphetModel(), horizon=14)
    folds = [
        FoldMask(
            first_train_timestamp=None,
            last_train_timestamp=pd.Timestamp("2019-10-19"),
            target_timestamps=[pd.Timestamp("2019-10-26"), pd.Timestamp("2019-10-27"), pd.Timestamp("2019-11-01")],
        ),
        FoldMask(
            first_train_timestamp=None,
            last_train_timestamp=pd.Timestamp("2019-11-02"),
            target_timestamps=[pd.Timestamp("2019-11-06"), pd.Timestamp("2019-11-08"), pd.Timestamp("2019-11-10")],
        ),
        FoldMask(
            first_train_timestamp=None,
            last_train_timestamp=pd.Timestamp("2019-11-16"),
            target_timestamps=[pd.Timestamp("2019-11-20"), pd.Timestamp("2019-11-22"), pd.Timestamp("2019-11-24")],
        ),
    ]
    metrics_df, forecast_df, fold_info_df = pipeline.backtest(ts=ts, metrics=[MAE()], n_folds=folds)

    plot_backtest(forecast_df=forecast_df, ts=ts, history_len=50)
    plt.savefig("plot_masks_partial_history.png")


if __name__ == "__main__":
    main()
```

plot_no_history:
plot_no_history

plot_partial_history:
plot_partial_history

plot_all_history:
plot_all_history

plot_masks_partial_history:
plot_masks_partial_history

**Comment by Mr-Geekman** [comment]
Script for `plot_backtest_interactive`:
```python
import pandas as pd

from etna.analysis import plot_backtest_interactive
from etna.datasets import TSDataset
from etna.metrics import MAE
from etna.models import ProphetModel
from etna.pipeline import FoldMask
from etna.pipeline import Pipeline


def main():
    df = pd.read_csv("examples/data/example_dataset.csv")
    df = TSDataset.to_dataset(df)
    ts = TSDataset(df, freq="D")

    pipeline = Pipeline(model=ProphetModel(), horizon=14)
    metrics_df, forecast_df, fold_info_df = pipeline.backtest(ts=ts, metrics=[MAE()], n_folds=3)

    # no history
    fig = plot_backtest_interactive(forecast_df=forecast_df, ts=ts, history_len=0)
    fig.write_image("plot_interactive_no_history.png")

    # partial history
    fig = plot_backtest_interactive(forecast_df=forecast_df, ts=ts, history_len=50)
    fig.write_image("plot_interactive_partial_history.png")

    # all history
    fig = plot_backtest_interactive(forecast_df=forecast_df, ts=ts, history_len="all")
    fig.write_image("plot_interactive_all_history.png")

    # fold mask example
    pipeline = Pipeline(model=ProphetModel(), horizon=14)
    folds = [
        FoldMask(
            first_train_timestamp=None,
            last_train_timestamp=pd.Timestamp("2019-10-19"),
            target_timestamps=[pd.Timestamp("2019-10-26"), pd.Timestamp("2019-10-27"), pd.Timestamp("2019-11-01")],
        ),
        FoldMask(
            first_train_timestamp=None,
            last_train_timestamp=pd.Timestamp("2019-11-02"),
            target_timestamps=[pd.Timestamp("2019-11-06"), pd.Timestamp("2019-11-08"), pd.Timestamp("2019-11-10")],
        ),
        FoldMask(
            first_train_timestamp=None,
            last_train_timestamp=pd.Timestamp("2019-11-16"),
            target_timestamps=[pd.Timestamp("2019-11-20"), pd.Timestamp("2019-11-22"), pd.Timestamp("2019-11-24")],
        ),
    ]
    metrics_df, forecast_df, fold_info_df = pipeline.backtest(ts=ts, metrics=[MAE()], n_folds=folds)

    fig = plot_backtest_interactive(forecast_df=forecast_df, ts=ts, history_len=50)
    fig.write_image("plot_interactive_masks_partial_history.png")


if __name__ == "__main__":
    main()
```

plot_interactive_no_history:
plot_interactive_no_history

plot_interactive_partial_history:
plot_interactive_partial_history

plot_interactive_all_history:
plot_interactive_all_history

plot_interactive_masks_partial_history:
plot_interactive_masks_partial_history

File: etna/analysis/plotters.py [hunk 1]
```diff
@@ -264,62 +288,108 @@ def plot_backtest(
...
+                # draw points on test
+                ax[i].scatter(backtest_df_slice_fold.index, backtest_df_slice_fold.target, color=lines_colors["test"])
+
+                # draw forecast
+                ax[i].scatter(
+                    forecast_df_slice_fold.index, forecast_df_slice_fold.target, color=lines_colors["forecast"]
+                )
+
+            # draw borders of current fold
+            opacity = 0.15 * ((fold_number + 1) % 2)
```
Comment (Ama16) [inline etna/analysis/plotters.py:369]: Maybe opacity = 0.15 * ((fold_number + 1) % 2) + 0.15?
In plot_masks_partial_history when even folds are located between non-folds, the background does not change. it is hard to watch

File: etna/analysis/plotters.py [hunk 2]
```diff
@@ -264,62 +288,108 @@ def plot_backtest(
...
+    fold_numbers = forecast_df[segments[0]]["fold_number"]
+    _validate_intersecting_segments(fold_numbers)
+    folds = sorted(set(fold_numbers))
 
+    # prepare dataframes
+    df = ts.df
     forecast_start = forecast_df.index.min()
     history_df = df[df.index < forecast_start]
     backtest_df = df[df.index >= forecast_start]
+    freq_timedelta = df.index[1] - df.index[0]
```
[the remark this answers is not in the document]
Comment (Mr-Geekman) [reply etna/analysis/plotters.py:320]: It isn't working with all freqs:
```python
pd.to_timedelta("D")
```
I have an error in such case.

[the remark this answers is not in the document]
Comment (Ama16) [reply etna/analysis/plotters.py:320]: hmm, okay then. It just doesn't look very good this way but ok

**Review by Ama16 — APPROVED** [review]

**Pull request merged**